api/v1/db/json/notes-dao.js

```diff
@@ -136,15 +172,38 @@ const postNotes = body => new Promise((resolve, reject) => {
     newNote.dateCreated = moment().toISOString();
     newNote.lastModified = newNote.dateCreated;
 
-    const noteFilePath = `${studentDir}/${noteID}.json`;
-    writeJSONFile(noteFilePath, newNote, { flag: 'wx' });
+    writeNote(noteID, newNote, true);
     incrementCounter(counterFilePath);
 
-    const serializedNote = getNoteByID(noteID);
-    resolve(serializedNote);
+    resolve(getNoteByID(noteID));
+  } catch (err) {
+    reject(err);
+  }
+});
+
+/**
+ * @summary Patch a note by noteID
+ * @function
+ * @param noteID
+ * @param body
+ * @returns {Promise} Promise object that represents the patched note
+ */
+const patchNoteByID = (noteID, body) => new Promise((resolve, reject) => {
+  try {
+    const rawNote = fetchNote(noteID);
+    if (!rawNote) {
+      resolve(null);
+    }
+
+    const { note, permissions } = body;
```

Comment:
Like the comment I left in the POST method, we should also validate the body for PATCH method as well. I imagine the solution should be pretty similar and probably can be shared for these two methods, so feel free to ignore this comment if you can fix it in the POST PR.

Reply:
This should be fixed now here https://github.com/osu-mist/notes-api/blob/feature/CO-1329-implement-patch-notes/openapi.yaml#L142